javascript/javascript2/week1/index.js

```diff
@@ -0,0 +1,141 @@
+    }, {
+        animal:"Dog",
+        description:"To have one as your spirit animal is a warning to be aware of hidden issues you haven’t come to know just yet; it’s also a reminder to possess a high level of loyalty with the people in your life, just like the animal does."
+    }, {
+        animal:"Elephant",
+        description:"Not only do elephants never forget, but they are highly intelligent mammals who symbolize honor, royalty, strength and power. "
+    }, {
+        animal:"Frog",
+        description:"We usually see frogs hanging out by bodies of water, and symbolize cleansing, rebirth, and peace."
+    }];
+
+
+let inputValue = '';
+
+input.addEventListener("input", onInputPress);
+btnSubmit.addEventListener("click", getRandomSpiritAnimal, { once: true });
+btnReset.addEventListener("click", getNewRandomSpiritAnimal);
+
+
+function onInputPress(event) {
+    // inputValue = event.target.value;
+    inputValue = input.value;
+    console.log(inputValue);
+  return inputValue;
+};
+
+
+function getRandomSpiritAnimal() {
+    if (!inputValue) {
+        return output.textContent = "Need the name";
+    };
+    const randNum = Math.floor(Math.random() * 10) + 1;
+    const randomAnimal = spiritAnimals[randNum];
+    const { animal, description } = randomAnimal;
+    output.textContent = ` ${inputValue} - ${animal}. ${description}`;
+    
+};
+
+function getNewRandomSpiritAnimal() {
+     if (!inputValue) {
+        return output.textContent = "Need the name";
+    };
+    const randNum = Math.floor(Math.random() * 10) + 1;
+    const randomAnimal = spiritAnimals[randNum];
+    const { animal, description } = randomAnimal;
+    output.textContent = ` ${inputValue} - ${animal}. ${description}`;
+};
+
+
+
+
+
+// 1.1 Find the shortest word
+
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+
+function findingShortestWord(arr) {
+    return arr.reduce((a, b) => a.length <= b.length ? a : b);
+}
+
+// console.log(findingShortestWord(danishWords));
+// console.log(danishWords)
+
+// 1.2. Find and count the Danish letters
+
+const danishChars = ["å", "ø", "æ"];
+let sum = 0;
+let sumÅ = 0;
+let sumØ = 0;
+let sumÆ = 0;
+
+
+function countDanishChars(string) {
+    const stringToLowerCase = string.toLowerCase();
+    for (let index = 0; index < stringToLowerCase.length; index++) {
+    const element = string[index];
+  
+    for (let index = 0; index < danishChars.length; index++) {
+        const element2 = danishChars[index];
+      
+        if (element === element2) {
+            sum += 1;
+            if (element === danishChars[0]) {
+                sumÅ += 1;
+            }
+            if (element === danishChars[1]) {
+                sumØ += 1;
+            }
+            if (element === danishChars[2]) {
+                sumÆ += 1;
+            }
+        }
+
+    }
+    }
+    return {total: sum, æ: sumÆ, ø: sumØ, å: sumÅ};
```

Comment:
Good solution! For a solution that requires a bit less code, Javascript's `.match()` function can be used. You can read in the first answer on this page how it can be used :) https://stackoverflow.com/questions/4009756/how-to-count-string-occurrence-in-string